packages/nodejs/src/__tests__/transmitter.test.ts

```diff
@@ -0,0 +1,168 @@
+          api_key: "",
+          name: "",
+          environment: "test",
+          hostname: "",
+          ...query
+        })
+        .reply(200, responseBody)
+    }
+
+    async function expectSampleResponse(
+      response: Promise<any>,
+      expectedBody: object = { json: "response" }
+    ) {
+      await expect(response).resolves.toEqual({
+        status: 200,
+        body: expectedBody
+      })
+    }
+
+    function mockHTTPSRequest() {
+      return jest.spyOn(https, "request").mockImplementation(opts => {
+        const clientRequestMock = {
+          on: () => {},
+          write: () => {},
+          end: () => {}
+        }
+
+        return (clientRequestMock as unknown) as ClientRequest
+      })
+    }
+
+    it("resolves to a response on success", async () => {
+      const transmitter = sampleTransmitter()
+
+      mockSampleRequest()
+
+      await expectSampleResponse(transmitter.transmit())
+    })
+
+    it("resolves to an empty object if the response is not JSON", async () => {
+      const transmitter = sampleTransmitter()
+
+      mockSampleRequest({ responseBody: "not JSON" })
+
+      await expectSampleResponse(transmitter.transmit(), {})
+    })
+
+    it("rejects on error", async () => {
+      const transmitter = sampleTransmitter()
+
+      // do not mock the request -- beforeAll uses nock to disable network
+      // requests, so any request to a non-mocked host will throw an error
+
+      await expect(transmitter.transmit()).rejects.toMatchObject({
+        error: expect.any(Error)
+      })
+    })
+
+    it("uses config values in the query string", async () => {
+      process.env.APPSIGNAL_PUSH_API_KEY = "some_api_key"
+      process.env.APPSIGNAL_APP_NAME = "myApp"
+      ;(process.env as any).NODE_ENV = "development"
+      process.env.APPSIGNAL_HOSTNAME = "myHostname"
+
+      const transmitter = sampleTransmitter()
+
+      mockSampleRequest({
+        query: {
+          api_key: "some_api_key",
+          name: "myApp",
+          environment: "development",
+          hostname: "myHostname"
+        }
+      })
+
+      await expectSampleResponse(transmitter.transmit())
+    })
+
+    it("uses the CA file from the config", async () => {
+      // disable nock, so we can mock the https library ourselves
+      nock.restore()
+
+      process.env.APPSIGNAL_CA_FILE_PATH = "/foo/bar"
+
+      jest.spyOn(fs, "accessSync").mockImplementation(() => {})
+
+      const fsReadFileSyncMock = jest
+        .spyOn(fs, "readFileSync")
+        .mockImplementation(() => "ca file contents")
+
+      const httpsRequestMock = mockHTTPSRequest()
+
+      sampleTransmitter().transmit()
+
+      expect(fsReadFileSyncMock).toHaveBeenCalledWith("/foo/bar", "utf-8")
+
+      expect(httpsRequestMock).toHaveBeenCalledWith(
+        expect.objectContaining({ ca: "ca file contents" }),
+        expect.anything()
+      )
+    })
+
+    it("uses the system CAs if the CA file is not readable", () => {
```

Comment:
```suggestion
    it("doesn't configure the CA if the CA file is not readable", () => {
```

At first I read this like we would set the CA with the system CAs if the file is not readable. The description is what ultimately happens, but that's out of scope of our package.